HW/iris.py

```diff
@@ -1,75 +1,142 @@
+#!/usr/bin/env python3
+
+import collections
+import os
+import sys
+
 import numpy as np
 import pandas as pd
 import plotly.express as px
 from sklearn.ensemble import RandomForestClassifier
 from sklearn.metrics import accuracy_score, classification_report, confusion_matrix
 from sklearn.model_selection import train_test_split
 from sklearn.neighbors import KNeighborsClassifier
+from sklearn.pipeline import Pipeline
 from sklearn.preprocessing import StandardScaler
 
+collections.Callable = collections.abc.Callable
+
+
 # importing dataset as a pandas DF
 # no header in the dataset so we define the headers manually
-iris_data = pd.read_csv(
-    "/Users/Maria/Downloads/iris.data",
-    header=None,
-    names=["sepal_length", "sepal_width", "petal_length", "petal_width", "class"],
-)
-# changing pandas df to numpy array
-iris_data_numpy = iris_data.to_numpy()
-
-# calculating summary statistics using numpy
-print("maximum value for each attribute : ", np.max(iris_data_numpy[:, 0:3], axis=0))
-print("minimum value for each attribute : ", np.min(iris_data_numpy[:, 0:3], axis=0))
-print("Average value for each attribute : ", np.mean(iris_data_numpy[:, 0:3], axis=0))
-print("Median value for each attribute : ", np.median(iris_data_numpy[:, 0:3], axis=0))
+def import_iris_data():
+    dirname = os.path.dirname(__file__)
+    iris_data_path = os.path.join(dirname, "./iris.data")
+    iris_data = pd.read_csv(
```

Comment:
This is working now without me having to download the dataset or modify anything